src/learning.jl

```diff
@@ -2,35 +2,35 @@ function Distributions.fit(::Type{BayesNet}, data::DataFrame, dag::DAG, cpd_type
 
     length(cpd_types) == nv(dag) || throw(DimensionMismatch("dag and cpd_types must have the same length"))
 
-    cpds = Array{CPD}(length(cpd_types))
+    cpds = Array{CPD}(undef, length(cpd_types))
     tablenames = names(data)
     for (i, target) in enumerate(tablenames)
         C = cpd_types[i]
-        parents = tablenames[in_neighbors(dag, i)]
+        parents = tablenames[inneighbors(dag, i)]
         cpds[i] = fit(C, data, target, parents)
     end
 
     BayesNet(cpds)
 end
-function Distributions.fit{C<:CPD}(::Type{BayesNet}, data::DataFrame, dag::DAG, ::Type{C})
+function Distributions.fit(::Type{BayesNet}, data::DataFrame, dag::DAG, ::Type{C}) where {C<:CPD}
 
-    cpds = Array{C}(nv(dag))
+    cpds = Array{C}(undef, nv(dag))
     tablenames = names(data)
     for (i, target) in enumerate(tablenames)
-        parents = tablenames[in_neighbors(dag, i)]
+        parents = tablenames[inneighbors(dag, i)]
         cpds[i] = fit(C, data, target, parents)
     end
 
     BayesNet(cpds)
 end
-Distributions.fit{T<:CPD}(::Type{BayesNet{T}}, data::DataFrame, dag::DAG) = fit(BayesNet, data, dag, T)
+Distributions.fit(::Type{BayesNet{T}}, data::DataFrame, dag::DAG) where {T<:CPD} = fit(BayesNet, data, dag, T)
 
 function _get_dag(data::DataFrame, edges::Tuple{Vararg{Pair{NodeName, NodeName}}})
     varnames = names(data)
     dag = DAG(length(varnames))
     for (a,b) in edges
-        i = findfirst(varnames, a)
-        j = findfirst(varnames, b)
+        i = something(findfirst(isequal(a), varnames), 0)
+        j = something(findfirst(isequal(b), varnames), 0)
         add_edge!(dag, i, j)
```

Comment:
This is going to fail even if i and j are set to 0. Might as well drop the `something` calls.